sdk/core/azure-core-amqp/src/impl/rust_amqp/amqp/private/management_impl.hpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "azure/core/amqp/internal/connection.hpp"
+#include "azure/core/amqp/internal/management.hpp"
+#include "azure/core/amqp/internal/session.hpp"
+#include "connection_impl.hpp"
+#include "message_receiver_impl.hpp"
+#include "message_sender_impl.hpp"
+#include "session_impl.hpp"
+
+#include <azure/core/credentials/credentials.hpp>
+
+#if ENABLE_UAMQP
+#include <azure_uamqp_c/amqp_management.h>
+#endif
+
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Amqp { namespace _detail {
+
+  class ManagementClientFactory final {
+  public:
+    static Azure::Core::Amqp::_internal::ManagementClient CreateFromInternal(
+        std::shared_ptr<ManagementClientImpl> clientImpl)
+    {
+      return Azure::Core::Amqp::_internal::ManagementClient(clientImpl);
+    }
+
+    static std::shared_ptr<ManagementClientImpl> GetImpl(
+        Azure::Core::Amqp::_internal::ManagementClient const& client)
+    {
+      return client.m_impl;
+    }
+  };
+
+  class ManagementClientImpl final : public ::std::enable_shared_from_this<ManagementClientImpl> {
+  public:
+    ManagementClientImpl(
+        std::shared_ptr<SessionImpl> session,
+        std::string const& managementEntityName,
+        _internal::ManagementClientOptions const& options,
+        _internal::ManagementClientEvents* managementEvents);
+
+    ~ManagementClientImpl() noexcept;
+
+    /**
+     * @brief Open the management instance.
+     *
+     * @returns A tuple consisting of the status code for the open and the description of the
+     * status.
+     */
+    _internal::ManagementOpenStatus Open(Context const& context = {});
```

Comment:
Comment/source mismatch. I see only the status code being returned, not a tuple.